### ISSUE TITLE
Unable to Update homebridge-config-ui-x from v4.56.4 → v4.58.0

### ISSUE THREAD

**Issue by TweedHat** [oznu/homebridge-config-ui-x#2187]
### Current Situation

I've been having issues updating this for weeks.  Any ideas on what might be wrong?

### Logs

DIR: /usr/local
CMD: npm install -g homebridge-config-ui-x@4.58.0

npm error code EACCES
npm error syscall rename
npm error path /usr/local/lib/node_modules/homebridge-config-ui-x
npm error dest /usr/local/lib/node_modules/.homebridge-config-ui-x-3gtnOIj4
npm error errno -13
npm error Error: EACCES: permission denied, rename '/usr/local/lib/node_modules/homebridge-config-ui-x' -> '/usr/local/lib/node_modules/.homebridge-config-ui-x-3gtnOIj4'
npm error     at async Object.rename (node:internal/fs/promises:782:10)
npm error     at async moveFile (/usr/local/lib/node_modules/npm/node_modules/@npmcli/fs/lib/move-file.js:30:5)
npm error     at async Promise.allSettled (index 0)
npm error     at async [reifyPackages] (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:325:11)
npm error     at async Arborist.reify (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:142:5)
npm error     at async Install.exec (/usr/local/lib/node_modules/npm/lib/commands/install.js:150:5)
npm error     at async Npm.exec (/usr/local/lib/node_modules/npm/lib/npm.js:207:9)
npm error     at async module.exports (/usr/local/lib/node_modules/npm/lib/cli/entry.js:74:5) {
npm error   errno: -13,
npm error   code: 'EACCES',
npm error   syscall: 'rename',
npm error   path: '/usr/local/lib/node_modules/homebridge-config-ui-x',
npm error   dest: '/usr/local/lib/node_modules/.homebridge-config-ui-x-3gtnOIj4'
npm error }
npm error
npm error The operation was rejected by your operating system.
npm error It is likely you do not have the permissions to access this file as the current user
npm error
npm error If you believe this might be a permissions issue, please double-check the
npm error permissions of the file and its containing directories, or try running
npm error the command again as root/Administrator.
npm error A complete log of this run can be found in: /Users/matthewm1minihome/.npm/_logs/2024-09-19T02_36_00_565Z-debug-0.log

Cleaning up npm cache, please wait...
npm cache cleared, please try updating homebridge-config-ui-x again.

Operation failed with code 243.
You can download this log file for future reference.
See https://github.com/homebridge/homebridge-config-ui-x/wiki/Troubleshooting for help.

### Homebridge UI Version

4.56.4

### Homebridge Version

7A19

### Node.js Version

v20.17.0 

### Operating System

macOS

### Environment Info

None

### Raspberry Pi Model

None

**Comment by NorthernMan54** [comment]
The log file shows the issue, looks like the /usr/local/lib/node_modules/ directory is not owned by the user running homebridge

**Comment by TweedHat** [comment]
Thank you for your help so far.  Any idea how I might be able to fix it?  This is on a mac mini with only 1 user, myself.  I'm guessing some update I did to MacOS broke this permission somehow?  I have been researching but havent figured out how to fix yet, I even tried to do a clean install, etc.

**Comment by bwp91** [comment]
Try running this command in the terminal

```
sudo chown -R $(whoami) $(npm config get prefix)/{lib/node_modules,bin,share}
```

run the command exactly as it is - there is nothing that needs replacing. it will likely ask you to enter your mac user password which is fine.

From here: https://elvisciotti.medium.com/mac-osx-fix-npm-eacces-permission-denied-7ed4b5368460

**Comment by PoppyS33d** [comment]
Just wanted to say thank you for this. Had the exact same issue and was just now able to update successfully following that command. Thanks @bwp91 !
